### ISSUE TITLE
Editing soruce code for Tiptap RTE doesn't work

### ISSUE THREAD

**Issue by stefanstankovic** [umbraco/Umbraco-CMS#17832]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

15.1.1

### Bug summary

I'm trying to replace TinyMCE in my project with a new solution Tiptap. First I want to test it so I created a new DataType for RTE that uses Tiptap.

I put an example text in the RTE that has a Heading and 3 paragraphs and published it. Then I tried to add a custom CSS class to the Heading directly in the source code. Opening the source code works and I can write the custom class, but after submitting the content and saving the class disappeared.

### Steps to reproduce

1. Create a new Data Type for RTE by using Tiptap
2. Create a new Element Document Type with a property that will use the new Data Type
3. Create another Element type that will represent the layout in the Block Grid
4. Create a new Data Type for the Block grid 
5. Add elements from steps 2 and 3 in the Grid and allow the element from step 3 to be added in the root and the element from step 2 to be added as part of element 3.
6. Create a Content page type that has the Block Grid and can be added to the root
7. Create a new Content page by using the Content page type created in step 5 and add the layout element and the element with Tiptap.
8. Add some content in the RTE and save it
9. Try to edit the source code

Editis source code doesn't work. You can change and submit changes but they won't be saved

**Comment by NguyenThuyLan** [comment]
Thank you @stefanstankovic  for reporting this issue.
I was able to reproduce it on v15.1.1, it is not able to add class or id into the HTML tag by editing source code mode using Tiptap RTE, after clicking submit, it is removed from the HTML tag. 
**Short steps to reproduce :**
1. Create a document type with Tiptap RTE
2. Create a content page and add a text string to the RTE
3. Click Edit source code of RTE, add a custom class/id into the HTML tag, then click submit
4. Open Edit source of RTE again, => custom class/id added is disappear

**Comment by ssplyuvrajsingh** [comment]
@leekelleher 
Yes, the same issue is here. When I add a span tag and some other tags by editing the source code, it replaces the previous one after I click on the submit button.

**Comment by iOvergaard** [comment]
Will be resolved in 15.3 with #18124
